File: src/main/java/au/id/micolous/metrodroid/transit/smartrider/SmartRiderTrip.java
package au.id.micolous.metrodroid.transit.smartrider;

import android.os.Parcel;
import android.support.annotation.Nullable;

import au.id.micolous.metrodroid.transit.Transaction;
import au.id.micolous.metrodroid.transit.TransactionTrip;
import au.id.micolous.metrodroid.transit.TransitCurrency;

class SmartRiderTrip extends TransactionTrip {
    SmartRiderTrip(Transaction el) {
        super(el);
    }

    protected SmartRiderTrip(Parcel in) {
        super(in);
    }

    public static final Creator<SmartRiderTrip> CREATOR = new Creator<SmartRiderTrip>() {
        @Override
        public SmartRiderTrip createFromParcel(Parcel in) {
            return new SmartRiderTrip(in);
        }

        @Override
        public SmartRiderTrip[] newArray(int size) {
            return new SmartRiderTrip[size];
        }
    };

    @Nullable
    @Override
    public TransitCurrency getFare() {
        int cost = 0;
        if (mStart != null)
            cost += ((SmartRiderTagRecord) mStart).getCost();
        if (mEnd != null)
            cost += ((SmartRiderTagRecord) mEnd).getCost();
        return TransitCurrency.AUD(cost);
    }

    @Override
    public String getRouteLanguage() {
        return "en-AU";
    }
}

File: src/commonMain/kotlin/au/id/micolous/metrodroid/transit/smartrider/SmartRiderTrip.kt
package au.id.micolous.metrodroid.transit.smartrider

import au.id.micolous.metrodroid.multi.Parcelize
import au.id.micolous.metrodroid.transit.*

@Parcelize
internal class SmartRiderTrip (override val capsule: TransactionTripCapsule): TransactionTripAbstract() {
    override val fare: TransitCurrency?
        get() = TransitCurrency.AUD(((start as? SmartRiderTagRecord?)?.cost ?: 0) +
                    ((end as? SmartRiderTagRecord?)?.cost ?: 0))

    override val routeLanguage: String?
        get() = "en-AU"

    constructor(el: Transaction) : this(TransactionTripAbstract.makeCapsule(el))
}
